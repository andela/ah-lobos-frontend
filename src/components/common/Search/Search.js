import React, { Component } from "react";
import propTypes from "prop-types";
import { connect } from "react-redux";
import { Form } from "semantic-ui-react";
import { searchMethod } from "../../../redux/actions/SearchAction";
import searchIm from "../../../assets/images/search.svg";

export class SearchItems extends Component {
  constructor() {
    super();
    this.state = {
      data: "",
      searchOption: ""
    };
  }

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  onSubmit = e => {
    e.preventDefault();
    const { data, searchOption } = this.state;
    this.props.searchMethod(searchOption, data);
    this.props.all.history.push("/search/content");
  };

  render() {
    return (
      <>
        <Form
          id="search"
          onSubmit={this.onSubmit}
          className="searchFunctionality"
        >
          <div className="searchDiv-container">
            <div className="searchDiv-input">
              <input
                type="text"
                name="data"
                className="searchIn"
                placeholder="Search..."
                onChange={this.onChange}
                value={this.state.data}
              ></input>
            </div>
            <div className="custom-select">
              <select name="searchOption" onChange={this.onChange}>
                <option value="all">all</option>
                <option value="title">title</option>
                <option value="author">author</option>
                <option value="description">description</option>
                <option value="tag">tag</option>
              </select>
            </div>
            <div className="searchDiv-button">
              <button className="button-search">
                <img src={searchIm} alt=""></img>
              </button>
            </div>
          </div>
        </Form>
      </>
    );
  }
}

SearchItems.propTypes = {
  searchMethod: propTypes.func.isRequired,
  all: propTypes.any
};

const mapStateToProps = state => ({
  searchData: state.searchData
});

export default connect(
  mapStateToProps,
  { searchMethod }
)(SearchItems);
