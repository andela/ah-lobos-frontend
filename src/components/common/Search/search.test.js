/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { shallow } from "enzyme";
import { SearchItems } from "./Search";

// const props = {
//   email: {},
//   setEmail: jest.fn(),
//   passwords: {},
//   confirmPassword: {},
//   setNewPassword: jest.fn(),
//   match: {
//     params: {
//       token: ""
//     }
//   }
// };

describe("Reset component", () => {
  const component = shallow(<SearchItems />);
  it("renders without errors", () => {
    expect(component).toMatchSnapshot();
  });
  // it("renders without error", () => {
  //   expect(componentNew).toMatchSnapshot();
  // });
  // it("renders without errors new password", () => {
  //   const wrapper = componentNew.find(".resetButton");
  //   expect(wrapper.length).toBe(1);
  // });
  // it("should call onSubmit method when the button is clicked", () => {
  //   const spy = jest.spyOn(component.instance(), "onSubmit");
  //   component.instance().forceUpdate();

  //   const Event = { preventDefault: () => {} };
  //   const form = component.find(".reset");
  //   form.simulate("submit", Event);
  //   expect(form.length).toBe(1);
  //   expect(spy).toHaveBeenCalled();
  // });
  it("should call onSubmit method when the button is clicked search", () => {
    const spy = jest.spyOn(component.instance(), "onSubmit");
    component.instance().forceUpdate();
    const Event = { preventDefault: () => {} };
    const form = component.find(".searchFunctionality");
    form.simulate("submit", Event);
    expect(form.length).toBe(1);
    expect(spy).toHaveBeenCalled();
  });
  // it("should call onChange method when the input value is changed", () => {
  //   const spy = jest.spyOn(component.instance(), "onChange");
  //   component.instance().forceUpdate();

  //   const event = {
  //     target: { value: "manzif60@gmail.com" }
  //   };

  //   const input = component.find(".thisButton").at(0);
  //   input.simulate("change", event);
  //   expect(input.length).toBe(1);
  //   expect(spy).toHaveBeenCalled();
  // });
  it("should call onChange method when the input change on search", () => {
    const spy = jest.spyOn(component.instance(), "onChange");
    component.instance().forceUpdate();

    const event = {
      target: { data: "Password12" }
    };

    const input = component.find(".searchIn").at(0);
    input.simulate("change", event);
    expect(input.length).toBe(1);
    expect(spy).toHaveBeenCalled();
  });
});
