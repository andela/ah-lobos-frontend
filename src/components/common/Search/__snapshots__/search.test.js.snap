// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset component renders without errors 1`] = `ShallowWrapper {}`;
